api/routes/roles/post.js

```diff
@@ -0,0 +1,73 @@
+const defaultRoleModel = require('../../db').models.role;
+const defaultActivityModel = require('../../db').models.activity;
+const can = require('../../auth/middleware').can;
+
+const validateRole = async (role, RoleModel, ActivityModel) => {
+  if (!role) {
+    throw new Error('Role must be specified');
+  }
+  if (!role.name) {
+    throw new Error('Role must have a name');
+  }
+  if (await RoleModel.where({ name: role.name }).fetch()) {
+    throw new Error(`Role name "${role.name}" already exists`);
+  }
+  if (!Array.isArray(role.activities)) {
+    throw new Error('Role must have a list of activities');
+  }
+  if (role.activities.filter(id => typeof id !== 'number').length > 0) {
+    throw new Error('Role activites must be numbers');
+  }
+
+  const validActivities = await ActivityModel.where(
+    'id',
+    'in',
+    role.activities
+  ).fetchAll({ columns: ['id'] });
+  const validIDs = validActivities.map(activity => activity.get('id'));
+  const invalidIDs = role.activities.filter(
+    roleID => !validIDs.includes(roleID)
+  );
+
+  if (invalidIDs.length) {
+    throw new Error(
+      `Activity IDs ${JSON.stringify(invalidIDs).replace(
+        /\[(.+)\]/,
+        '$1'
+      )} are invalid`
+    );
+  }
+
+  return {
+    name: role.name,
+    activities: role.activities.filter(id => !Number.isNaN(Number(id)))
```

Comment:
is this filter necessary after the validation check on #18 checking for numbers only?

Reply:
😂 Nope.  How silly.  I'll remove in a future PR - I need to run `prettier` on the whole API codebase.  I've finally setup Atom to run it automatically on every save so hopefully it won't be a big deal going forward, but still need to clean up the past...  😬

Reply:
nice - the atom-prettier on save hook is pretty 🤘